README.md
# Welcome to Twigs monorepo

## Packages
1. react-components (`@sparrowengg/twigs-react`) → React UI components 
2. react-icons (`@sparrowengg/twigs-react-icons`) → SVG Icons package built with React


## monorepo
This repo is built with lerna to support monorepo. commands can be run from the root folder, and will be applicable for all of the packages under the monorepo

### How do we handle monorepo
We've achieving mono-repo using [lerna](https://lerna.js.org/). You can read more about the advantages of using lerna [here](https://lerna.js.org/docs/introduction#why-lerna).

Under the hood, lerna makes use of workspaces API provided by package managers ([yarn](https://yarnpkg.com/features/workspaces), npm, pnpm)

### Why did we choose lerna?
1. packages can be independant and still consume each other like an npm package way. 
2. avoid installing the same npm dependancy for sub-packages. For eg: if 2 of our package use same npm dependancy, lerna will install at global level and reference to it. If a dependancy is only applicable for one of the sub-package, lerna will keep a copy at package level. 
3. lerna makes it easy to run commands across packages parallelly. For eg: you can run `test` or `dev` with one single command across packages. `build` command also can happen in parallel. lerna is smart enough to build package 1 before package 2, if package 2 has a dependancy of package 1. 
4. caching of files. Unless the files are changes, lerna will not put effort into re-building the same thing. This will improve the build time. 

## How to set-up

1. Clone the repo
2. install the dependencies using [yarn](https://yarnpkg.com/) `yarn install`

**Other scripts**

1. build `npx lerna run build`

2. build static storybook `npx lerna run build-storybook`

3. To delete all node_modules folders (even from subpackages), `npx lerna clean -y`

4. If you'd like to run command against one package alone, run `npx lerna run build --scope=package-name`

CHANGELOG.md
# Changelog

All notable changes to this project will be documented in this file.

The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/)
and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0.html).

Generated by [`auto-changelog`](https://github.com/CookPete/auto-changelog).

## [v0.0.17](https://bitbucket.org/surveysparrow/twigs-react/compare/v0.0.17..v0.0.9)

### Merged

- Components/toast tooltip [`#17`](https://bitbucket.org/surveysparrow/twigs-react/pull-requests/17)
- Modify avatar api [`#16`](https://bitbucket.org/surveysparrow/twigs-react/pull-requests/16)
- Components/select pills tags [`#15`](https://bitbucket.org/surveysparrow/twigs-react/pull-requests/15)
- Remove stories from dist [`#14`](https://bitbucket.org/surveysparrow/twigs-react/pull-requests/14)
- Complete stories [`#13`](https://bitbucket.org/surveysparrow/twigs-react/pull-requests/13)
- added types to most stories [`#12`](https://bitbucket.org/surveysparrow/twigs-react/pull-requests/12)
- Stories for components [`#11`](https://bitbucket.org/surveysparrow/twigs-react/pull-requests/11)
- add radio button and toggle switch component [`#10`](https://bitbucket.org/surveysparrow/twigs-react/pull-requests/10)
- add drawer component, update avatar component [`#9`](https://bitbucket.org/surveysparrow/twigs-react/pull-requests/9)

### Commits

- add chip component, update checkbox, grid, button components [`5808300`](https://bitbucket.org/surveysparrow/twigs-react/commits/580830088d50534f57eef73d73fbb3bab2891065)
- deleted example folder [`afa6417`](https://bitbucket.org/surveysparrow/twigs-react/commits/afa6417156c93863023dca7e0c59961b31bf876a)
- added toast component [`6dfc020`](https://bitbucket.org/surveysparrow/twigs-react/commits/6dfc020f81acae1c1d4c9fe2a5b6680e5df75e19)
- removed act from unit tests [`80ffb62`](https://bitbucket.org/surveysparrow/twigs-react/commits/80ffb629c25497db26f82249278ff60d29657bdb)
- update test case for chip, remove react-select dependency [`dedcb40`](https://bitbucket.org/surveysparrow/twigs-react/commits/dedcb40ee8d8909e3cf6d791dd4538c51d7c926b)
- updated button props and stories [`9033e74`](https://bitbucket.org/surveysparrow/twigs-react/commits/9033e7479be95bb7631651fdcde9f451478f0649)
- added tooltip component [`8c1c04a`](https://bitbucket.org/surveysparrow/twigs-react/commits/8c1c04a5ba9c28c2de5b9a57b936010e7c97878b)
- added story for remaining components [`5534e2c`](https://bitbucket.org/surveysparrow/twigs-react/commits/5534e2c846458fd696fb7fda7359442b71b66e72)
- generated changelog [`887298e`](https://bitbucket.org/surveysparrow/twigs-react/commits/887298e18a509253e49c398ae830611a3d69db3f)
- remove unwanted chip variants [`1368604`](https://bitbucket.org/surveysparrow/twigs-react/commits/1368604528a9b819f719e3dbbba0b6e29fd8b668)
- added icon-button component [`bcc8662`](https://bitbucket.org/surveysparrow/twigs-react/commits/bcc86627ecb2322b23ba73046167915413f1cce1)
- added dialog story [`1587574`](https://bitbucket.org/surveysparrow/twigs-react/commits/15875742655e72e5cea47e488668c6d18a0c8c15)
- added dropdown stories [`5ec5986`](https://bitbucket.org/surveysparrow/twigs-react/commits/5ec59864a9e055fc9507b8be8c8dde312a045dbd)
- modified avatar component [`ab0cd6e`](https://bitbucket.org/surveysparrow/twigs-react/commits/ab0cd6ec11ff58b7375925e48e475237aa6cf7d9)
- updated avatar props and tests [`a070fbd`](https://bitbucket.org/surveysparrow/twigs-react/commits/a070fbd1e54cd2ac143106e8bb079c0668cabc14)
- add missing as prop in grid, add tooltip content prop in storybook, add toast duration prop in storybook [`09158b0`](https://bitbucket.org/surveysparrow/twigs-react/commits/09158b0fd36c21dd4cc6573fc00d5df3caf92143)
- added drawer story [`7632075`](https://bitbucket.org/surveysparrow/twigs-react/commits/76320752f9f956dfdbe7f5d7fffa52dfb201dc64)
- package changes [`92e5924`](https://bitbucket.org/surveysparrow/twigs-react/commits/92e5924fe1f8ce0535b1c5570c70685732c9b7d8)
- rename props in chip [`9a36e22`](https://bitbucket.org/surveysparrow/twigs-react/commits/9a36e22105b68c109e01ac5882a06d5d5a77a2ab)
- update font, update themeprovider to call globalstyles on mount [`485793b`](https://bitbucket.org/surveysparrow/twigs-react/commits/485793b2d2d97c6340674c16463a03c1e75479cb)
- updated pipelines [`aadca7d`](https://bitbucket.org/surveysparrow/twigs-react/commits/aadca7d2c285d5f285fc15b105ef9153155d5091)
- remove isBordered variant for avatar, added default sub menu icon in dropdown [`f4a0256`](https://bitbucket.org/surveysparrow/twigs-react/commits/f4a0256854f17306dd12c5d6ea6b02541ea78f9e)
- added radio story [`4bd20e1`](https://bitbucket.org/surveysparrow/twigs-react/commits/4bd20e1e4c07bd4e0850729d6e9ec26eae48f8b1)
- added story for button [`fbcd76e`](https://bitbucket.org/surveysparrow/twigs-react/commits/fbcd76eb60f6471bd0afa746bde65f2b38f2c84e)
- added tab story [`e1ad4bf`](https://bitbucket.org/surveysparrow/twigs-react/commits/e1ad4bf49df7a1404eb4c05382a574cfd36298d7)
- added default variant for toast [`c7cea11`](https://bitbucket.org/surveysparrow/twigs-react/commits/c7cea112741e8843770cb6f9a2251808c37507b5)
- updated font sizes for avatar [`46a098a`](https://bitbucket.org/surveysparrow/twigs-react/commits/46a098a666f1debc841ee6ff8f621d6d1c872772)
- added story for chip [`15df8ee`](https://bitbucket.org/surveysparrow/twigs-react/commits/15df8ee4fd19f868c0ff53dd14efa31f52156473)
- added avatar story [`e6373d1`](https://bitbucket.org/surveysparrow/twigs-react/commits/e6373d1f8d32ffd6fe042fb7a8de4fe9ffd5fcf7)
- added text story [`9f12bb3`](https://bitbucket.org/surveysparrow/twigs-react/commits/9f12bb3489b708de88586249f3ee8ebe76d0e5a5)
- added border box for input component [`adb74c9`](https://bitbucket.org/surveysparrow/twigs-react/commits/adb74c99d931d3d69e5d68a64e021f0f352ccf9a)
- added checkbox story [`f4b0b08`](https://bitbucket.org/surveysparrow/twigs-react/commits/f4b0b083128e363e8b42c442fbc49842f4fd6090)

stitches.config.ts
import { createStitches } from '@stitches/react';
import type * as Stitches from '@stitches/react';

export const resetStyles = {
  'html, body, div, span, applet, object, iframe, h1, h2, h3, h4, h5, h6, p, blockquote, pre, a, abbr, acronym, address, big, cite, code, del, dfn, em, img, ins, kbd, q, s, samp, small, strike, strong, sub, sup, tt, var, b, u, i, center, dl, dt, dd, ol, ul, li, fieldset, form, label, legend, table, caption, tbody, tfoot, thead, tr, th, td, article, aside, canvas, details, embed, figure, figcaption, footer, header, hgroup, main, menu, nav, output, ruby, section, summary, time, mark, audio, video': {
    margin: '0',
    padding: '0',
    border: '0',
    fontSize: '100%',
    verticalAlign: 'baseline',
  },
  'article, aside, details, figcaption, figure, footer, header, hgroup, main, menu, nav, section': {
    display: 'block',
  },
  '*[hidden]': {
    display: 'none',
  },
  body: {
    lineHeight: '1',
  },
  'ol, ul': {
    listStyle: 'none',
  },
  'blockquote, q': {
    quotes: 'none',
  },
  'blockquote:before, blockquote:after, q:before, q:after': {
    content: '',
    // eslint-disable-next-line no-dupe-keys
    // content: 'none',
  },
  table: {
    borderSpacing: '0',
  },
};

export const defaultTheme = {
  colors: {
    primary: '#EF7D7D',
    secondary: '#56B0BB',
    text: '#3B3B3B',
    mood: '#F8F8F8',
    link: '#4A9CA6',
    tertiary50: '#F9F8F8',
    tertiary100: '#F2F0F0',
    tertiary200: '#E5E1E1',
    tertiary300: '#CCC5C5',
    tertiary400: '#A69B9B',
    tertiary500: '#9B8E8E',
    tertiary600: '#948686',
    tertiary700: '#807373',
    tertiary800: '#736868',
    tertiary900: '#665C5C',
    primary50: '#FFF6F6',
    primary100: '#FFECEB',
    primary200: '#FFDAD8',
    primary300: '#FAB5B3',
    primary400: '#EF7D7D',
    primary500: '#DA7272',
    primary600: '#C86767',
    primary700: '#B45D5D',
    primary800: '#A35353',
    primary900: '#904949',
    attention50: '#FFF6EF',
    attention100: '#FFEDE0',
    attention200: '#FFDCC2',
    attention300: '#F8B988',
    attention400: '#F3953F',
    attention500: '#E0893A',
    attention600: '#BB722F',
    attention700: '#AA6628',
    attention800: '#995C25',
    attention900: '#89511E',
    highlight50: '#FFFCDA',
    highlight100: '#FFF7AD',
    highlight200: '#FFF27D',
    highlight300: '#FFED4B',
    highlight400: '#FFE81A',
    highlight500: '#E6CF00',
    highlight600: '#B3A100',
    highlight700: '#807300',
    highlight800: '#786B03',
    highlight900: '#695F04',
    positive50: '#F4FAF1',
    positive100: '#E8F4E3',
    positive200: '#D4E8CA',
    positive300: '#A8D291',
    positive400: '#67B034',
    positive500: '#5EA130',
    positive600: '#55932A',
    positive700: '#4C8425',
    positive800: '#437720',
    positive900: '#3C691C',
    secondary50: '#F3F9FA',
    secondary100: '#E6F3F4',
    secondary200: '#CEE7EA',
    secondary300: '#9CCFD6',
    secondary400: '#56B0BB',
    secondary500: '#4A9CA6',
    secondary600: '#448E97',
    secondary700: '#3B8088',
    secondary800: '#36737A',
    secondary900: '#2E666D',
    system50: '#F3F9FF',
    system100: '#E5F2FF',
    system200: '#CFE4FF',
    system300: '#A0C9FF',
    system400: '#70ADFF',
    system500: '#3E90FD',
    system600: '#007AFF',
    system700: '#0472EE',
    system800: '#0A66D8',
    system900: '#005BC0',
    info50: '#F7F7FF',
    info100: '#EFEFFF',
    info200: '#E1E0FF',
    info300: '#C4C0FE',
    info400: '#9E93FA',
    info500: '#8F85E3',
    info600: '#8279D0',
    info700: '#756DBB',
    info800: '#6962A9',
    info900: '#5D5696',
    negative50: '#FFF6F3',
    negative100: '#FDEDE8',
    negative200: '#FFDAD0',
    negative300: '#FFB4A1',
    negative400: '#FA7659',
    negative500: '#F65633',
    negative600: '#E75030',
    negative700: '#D14729',
    negative800: '#BC4024',
    negative900: '#A9371E',
    neutral50: '#F8F8F8',
    neutral100: '#F1F1F1',
    neutral200: '#E2E2E2',
    neutral300: '#C6C6C6',
    neutral400: '#9E9E9E',
    neutral500: '#919191',
    neutral600: '#848484',
    neutral700: '#767676',
    neutral800: '#6A6A6A',
    neutral900: '#3B3B3B',
    black50: '#0000000A',
    black100: '#00000014',
    black200: '#0000001A',
    black300: '#00000026',
    black400: '#00000033',
    black500: '#0000004D',
    black600: '#00000080',
    black700: '#000000B2',
    black800: '#000000CC',
    black900: '#000000',
    white50: '#FFFFFF0D',
    white100: '#FFFFFF14',
    white200: '#FFFFFF1A',
    white300: '#FFFFFF26',
    white400: '#FFFFFF33',
    white500: '#FFFFFF4D',
    white600: '#FFFFFF80',
    white700: '#FFFFFFB2',
    white800: '#FFFFFFCC',
    white900: '#FFFFFF',
  },
  space: {
    1: '0.125rem',
    2: '0.25rem',
    3: '0.375rem',
    4: '0.5rem',
    5: '0.625rem',
    6: '0.75rem',
    7: '0.875rem',
    8: '1rem',
    9: '1.125rem',
    10: '1.25rem',
    11: '1.375rem',
    12: '1.5rem',
    13: '1.625rem',
    14: '1.75rem',
    15: '1.875rem',
    16: '2rem',
    17: '2.125rem',
    18: '2.25rem',
    19: '2.375rem',
    20: '2.5rem',
    21: '2.625rem',
    22: '2.75rem',
    23: '2.875rem',
    24: '3rem',
    25: '3.125rem',
    26: '3.25rem',
    27: '3.375rem',
    28: '3.5rem',
    29: '3.625rem',
    30: '3.75rem',
    31: '3.875rem',
    32: '4rem',
    33: '4.125rem',
    34: '4.25rem',
    35: '4.375rem',
    36: '4.5rem',
    37: '4.625rem',
    38: '4.75rem',
    39: '4.875rem',
    40: '5rem',
    41: '5.125rem',
    42: '5.25rem',
    43: '5.375rem',
    44: '5.5rem',
    45: '5.625rem',
    46: '5.75rem',
    47: '5.875rem',
    48: '6rem',
    49: '6.125rem',
    50: '6.25rem',
  },
  fontSizes: {
    xxs: '0.579rem',
    xs: '0.694rem',
    sm: '0.833rem',
    md: '1rem',
    lg: '1.2rem',
    xl: '1.44rem',
    '2xl': '1.728rem',
    '3xl': '2.074rem',
    '4xl': '2.488rem',
    '5xl': '2.986rem',
  },
  fonts: {
    body: 'sytem-ui',
    heading: 'sans-serif',
  },
  fontWeights: {
    1: '100',
    2: '200',
    3: '300',
    4: '400',
    5: '500',
    6: '600',
    7: '700',
    8: '800',
    9: '900',
  },
  lineHeights: {
    xxs: '0.75rem',
    xs: '1rem',
    sm: '1.25rem',
    md: '1.5rem',
    lg: '1.75rem',
    xl: '2rem',
    '2xl': '2.5rem',
    '3xl': '3rem',
    '4xl': '4rem',
  },
  letterSpacings: {},
  sizes: {
    1: '4px',
    2: '8px',
    3: '12px',
    4: '16px',
    5: '20px',
    6: '24px',
    7: '28px',
    8: '32px',
    9: '36px',
    10: '40px',
    11: '44px',
    12: '48px',
    13: '52px',
    14: '56px',
    15: '60px',
    16: '64px',
    17: '68px',
    18: '72px',
    19: '76px',
    20: '80px',
    21: '84px',
    22: '88px',
    23: '92px',
    24: '96px',
    25: '100px',
    26: '104px',
    27: '108px',
    28: '112px',
    29: '116px',
    30: '120px',
    31: '124px',
    32: '128px',
    33: '132px',
    34: '136px',
  },
  borderWidths: {
    xs: '1px',
    sm: '2px',
    md: '3px',
    lg: '4px',
    xl: '5px',
  },
  borderStyles: {},
  radii: {
    xs: '3px',
    sm: '5px',
    md: '8px',
    lg: '10px',
    xl: '15px',
    '2xl': '20px',
    '3xl': '30px',
    round: '50%',
    pill: '9999px',
  },
  shadows: {
    sm: '0px 5px 15px rgba(0, 0, 0, 0.04)',
  },
  zIndices: {},
  transitions: {
    1: '0.1s',
    2: '0.2s',
    3: '0.3s',
  },
};

export const {
  styled,
  theme,
  createTheme,
  globalCss,
  keyframes,
  config,
  reset,
  css,
} = createStitches({
  prefix: 'twigs',
  theme: defaultTheme,
  utils: {
    p: (value: Stitches.PropertyValue<'padding'>) => ({
      padding: value,
    }),
    pt: (value: Stitches.PropertyValue<'paddingTop'>) => ({
      paddingTop: value,
    }),
    pr: (value: Stitches.PropertyValue<'paddingRight'>) => ({
      paddingRight: value,
    }),
    pb: (value: Stitches.PropertyValue<'paddingBottom'>) => ({
      paddingBottom: value,
    }),
    pl: (value: Stitches.PropertyValue<'paddingLeft'>) => ({
      paddingLeft: value,
    }),
    px: (value: Stitches.PropertyValue<'paddingLeft'>) => ({
      paddingLeft: value,
      paddingRight: value,
    }),
    py: (value: Stitches.PropertyValue<'paddingTop'>) => ({
      paddingTop: value,
      paddingBottom: value,
    }),

    m: (value: Stitches.PropertyValue<'margin'>) => ({
      margin: value,
    }),
    mt: (value: Stitches.PropertyValue<'marginTop'>) => ({
      marginTop: value,
    }),
    mr: (value: Stitches.PropertyValue<'marginRight'>) => ({
      marginRight: value,
    }),
    mb: (value: Stitches.PropertyValue<'marginBottom'>) => ({
      marginBottom: value,
    }),
    ml: (value: Stitches.PropertyValue<'marginLeft'>) => ({
      marginLeft: value,
    }),
    mx: (value: Stitches.PropertyValue<'marginLeft'>) => ({
      marginLeft: value,
      marginRight: value,
    }),
    my: (value: Stitches.PropertyValue<'marginTop'>) => ({
      marginTop: value,
      marginBottom: value,
    }),
  },
});

export const globalStyles = globalCss({
  ...resetStyles,
  '*': { margin: 0, padding: 0, fontFamily: '$body' },
  '*, :before, :after': { boxSizing: 'border-box' },
});
